experimental/single_number_formatter/src/provider.rs

```diff
@@ -69,11 +69,10 @@ pub enum PatternSelection {
 
 // TODO(#3836): replace this with Option<PlaceHolder>, enum PlaceHolder { Index(usize), ISO }
 // and encapsulate the encoding in the ULE implementation.
-pub const USE_ISO_CODE: u16 = u16::MAX - 1;
-pub const NO_PLACE_HOLDER: u16 = u16::MAX;
+pub const NO_PLACE_HOLDER: u16 = 0b0011_1111_1111;
+pub const USE_ISO_CODE: u16 = 0b0011_1111_1110;
+pub const MAX_PLACE_HOLDER_INDEX: u16 = 0b0011_1111_1101;
```

Comment:
Thought: If we changed the ULE representation in the future to give more range to these, it would be useful to have the `u16` return values remain the same, and only squish them down to these lower values when reading or writing to the ULE. This would also avoid the need to change the JSON representation.

Reply:
This is basically tracked in the TODO at the top of this block.